### ISSUE TITLE
bug: several problems caused by "etcdserver: request timed out"

### ISSUE THREAD

**Issue by wangrunji0408** [risingwavelabs/risingwave#5103]
### Describe the bug

After #5084, we found several problems caused by "etcdserver: request timed out" on the simulator.

In the following tests, each etcd request may block for 5~15s and then get timeout error with a 1% probability.

### To Reproduce

Checkout 2dabf82, then:

```sh
MADSIM_TEST_SEED=<seed> ./risedev sslt --release -- --etcd-timeout-rate=0.01 "e2e_test/streaming/**/*.slt"
```

Note: The following seeds may only work on M1 Macbook Pro. If you are in another environment, the seed will be different.

[FIXED] MADSIM_TEST_SEED=2,3,4,16,19
```
thread '<unnamed>' panicked at 'Failed to connect to database: Error { kind: Connect, cause: Some(Custom { kind: ConnectionRefused, error: "connection refused" }) }', src/tests/simulation/src/main.rs:365:14
```

[FIXED] MADSIM_TEST_SEED=9
```
thread '<unnamed>' panicked at 'no entry found for key', src/meta/src/stream/scheduler.rs:98:23
```

MADSIM_TEST_SEED=1,15
```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: test error at e2e_test/streaming/datagen.slt:18: query result mismatch:
[SQL] select v1, v2 from s1 where v1 is not null limit 15;
[Diff]
1 11
10 20
2 12
3 13
4 14
5 15
6 16
7 17
8 18
9 19
```

MADSIM_TEST_SEED=17
```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: test error at e2e_test/streaming/datagen.slt:48: query result mismatch:
[SQL] select v1 from s1 order by v1 limit 10;
[Diff]
1
2
3
4
5
6
7
8
9
10
```

MADSIM_TEST_SEED=6,8,10,11,12,13,18
```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: test error at e2e_test/streaming/nexmark_endless.slt:10: query result mismatch:
[SQL] select count(*) > 0 as has_results from nexmark_q7;
[Diff]
t
f
```

MADSIM_TEST_SEED=5,7,14,20
```
Pass
```

_Originally posted by @wangrunji0408 in https://github.com/risingwavelabs/risingwave/issues/5084#issuecomment-1236641036_

**Comment by BugenZhao** [comment]
For `e2e_test/streaming/[datagen|nexmark_endless].slt`, the data are generated over time and the `sleep` is hard-coded, that is, we assume the results can be generated in reasonable time. However, the result might be incorrect if there're some exceptions that break this assumption. I think it's expected.

**Comment by wangrunji0408** [comment]
I see. Is there a way to make these tests fault-resilient?

**Comment by BugenZhao** [comment]
> Is there a way to make these tests fault-resilient?

I've no idea. :( Unless we have some SQL syntaxs for checking the source state.

> We'd better "resolve" this issue rather than simply "ignore" it. Because it's possible to happen in normal e2e tests too

Agree. It frequently happens these days.

**Comment by wangrunji0408** [comment]
The "connection refused" error is caused by a deadlock in meta #5089 which will be fixed by #5042.

**Comment by zwang28** [comment]
> The "connection refused" error is caused by a deadlock in meta https://github.com/risingwavelabs/risingwave/issues/5089 which will be fixed by https://github.com/risingwavelabs/risingwave/pull/5042.

The deadlock is [fixed](https://github.com/risingwavelabs/risingwave/pull/5126) in main.
